info.limpet.rcp/src/info/limpet/rcp/data_provider/data/ReflectivePropertySource.java

```diff
@@ -0,0 +1,174 @@
+package info.limpet.rcp.data_provider.data;
+
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.Status;
+import org.eclipse.ui.views.properties.IPropertyDescriptor;
+import org.eclipse.ui.views.properties.IPropertySource;
+import org.eclipse.ui.views.properties.TextPropertyDescriptor;
+
+import info.limpet.UIProperty;
+import info.limpet.rcp.Activator;
+import info.limpet.rcp.propertyeditors.SliderPropertyDescriptor;
+
+public class ReflectivePropertySource implements IPropertySource
+{
+
+	private final Object object;
+	private IPropertyDescriptor[] propertyDescriptors;
+	private Map<String, PropertyDescriptor> descriptorPerProperty;
+
+	public ReflectivePropertySource(Object object)
+	{
+		this.object = object;
+	}
+
+	@Override
+	public Object getEditableValue()
+	{
+		return object;
+	}
+
+	@Override
+	public IPropertyDescriptor[] getPropertyDescriptors()
+	{
+		if (propertyDescriptors == null)
+		{
+			initPropertyDescriptors(object.getClass());
+		}
+		return propertyDescriptors;
+	}
+
+	private void initPropertyDescriptors(Class<?> cls)
+	{
+
+		descriptorPerProperty = new HashMap<String, PropertyDescriptor>();
+
+		List<IPropertyDescriptor> result = new ArrayList<IPropertyDescriptor>();
+
+		try
+		{
+			PropertyDescriptor[] beanPropertyDescriptors = Introspector.getBeanInfo(
+					cls).getPropertyDescriptors();
+			for (PropertyDescriptor beanPropertyDescriptor : beanPropertyDescriptors)
+			{
+				UIProperty annotation = beanPropertyDescriptor.getReadMethod()
+						.getAnnotation(UIProperty.class);
+				if (annotation != null)
+				{
+
+					org.eclipse.ui.views.properties.PropertyDescriptor descriptor = null;
+					
+					String propId = beanPropertyDescriptor.getName();
+					String propName = annotation.name();
+					String propCategory = annotation.category();
+					
+					if (beanPropertyDescriptor.getWriteMethod() != null)
+					{
+						Class<?> propertyType = beanPropertyDescriptor.getPropertyType();
+						if (propertyType.equals(String.class))
+						{
+							descriptor = new TextPropertyDescriptor(
+									propId, propName);
+						}
+						else if ("boolean".equals(propertyType.getName()))
+						{
+							descriptor = new CheckboxPropertyDescriptor(
```

Comment:
@dinkoivanov  - do you know how to make the UI show a checkbox, instead of the *true* / *false* text?